src/mongocxx/database.cpp

```diff
@@ -97,6 +97,11 @@ class collection database::create_collection(stdx::string_view name,
     return mongocxx::collection(*this, static_cast<void*>(result));
 }
 
+void database::drop() {
+    bson_error_t error;
+    libmongoc::database_drop(_impl->database_t, &error);
```

Comment:
note that the docs for mongoc_database_drop state that the error will always be set if the return value is false.